### PR TITLE
Fix System.ComponentModel.Primitives NREs

### LINKED ISSUE #38997: new CategoryAttribute(null).IsDefaultAttribute throws NRE
Should return `false`

### LINKED ISSUE #38998: new CategoryAttribute(null).GetHashCode() throws NRE
Should return 0

### LINKED ISSUE #38999: new CategoryAttribute(null).Equals(new CategoryAttribute("category")) throws NRE
Should return false

### LINKED ISSUE #39009: new InvalidEnumArgumentException("argumentName", 1, null) throws NRE
Should be ANE

```cs
new InvalidEnumArgumentException("argumentName", 1, null)
```

### LINKED ISSUE #39010: new EventHandlerList().AddHandlers(null) throws NRE
Should be ANE

```cs
new EventHandlerList().AddHandlers(null)
```

### PR DESCRIPTION
**Pull request by hughbe**
Fixes #38999
Fixes #38998
Fixes #38997
Fixes #39009
Fixes #39010

### CODE REVIEW

**Comment by safern** [comment]
@hughbe thanks for your PR. I think this is great stuff to have. @wtgodbe @danmosemsft what is the BAR for this PR to make it into 3.0?

**Comment by wtgodbe** [comment]
I don't believe we want to take any more API breaking changes at this point - @danmosemsft can you confirm?

**Comment by safern** [comment]
IMHO I don’t think this is breaking, I don’t expect someone to catch a NRE and have behavior dependent on a NRE to be thrown.

I think this change is good to have.

**Comment by hughbe** [comment]
I audited the other attributes in the library for this sort of bug and fixed the remaining issues

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/InvalidEnumArgumentException.cs [hunk 1]
```diff
@@ -49,8 +49,12 @@ public InvalidEnumArgumentException(string argumentName, int invalidValue, Type
             : base(SR.Format(SR.InvalidEnumArgument,
                                 argumentName,
                                 invalidValue.ToString(CultureInfo.CurrentCulture),
```
Comment (hughbe) [inline src/System.ComponentModel.Primitives/src/System/ComponentModel/InvalidEnumArgumentException.cs]: nit: I don't think we actually need this `ToString` as `SR.Format` does this for us

**Comment by danmosemsft** [comment]
Fine by me

File: src/System.ComponentModel.Primitives/src/System/ComponentModel/InvalidEnumArgumentException.cs (hunk 1 above)
Comment (safern) [reply src/System.ComponentModel.Primitives/src/System/ComponentModel/InvalidEnumArgumentException.cs]: Right. Would you mind cleaning that up as part of this PR?

**Pull request merged**